artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageIndexCacheImpl.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.core.paging.cursor.impl;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.activemq.artemis.core.paging.PagedMessage;
+import org.apache.activemq.artemis.core.paging.cursor.PageCache;
+import org.apache.activemq.artemis.core.paging.impl.Page;
+import org.jboss.logging.Logger;
+
+public class PageIndexCacheImpl implements PageCache {
+   private static final Logger logger = Logger.getLogger(PageIndexCacheImpl.class);
+
+   private final long pageId;
+   private Page page;
+
+   // message number to file position map
+   private TreeMap<Integer, Integer> messageIndex;
```

Comment:
ditto, could this be primitive map, or even simply an array of ints, using the array idx as the key.